### PR TITLE
filesystem.hpp, add support for Haiku

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
Don't merge yet, need to see if I can find a way to link it to libnetwork on Haiku (for bind and socket).

**Comment by jessicah** [comment]
https://github.com/gulrak/filesystem/blob/master/cmake/GhcHelper.cmake#L48

Add something like:
```
if (HAIKU)
  target_link_libraries(${targetName} -lnetwork)
endif()
```

**Comment by jessicah** [comment]
You may also need for the second macro, depends which part(s) require libnetwork.

**Comment by Begasus** [comment]
@jessicah thanks for the tip, following the fix for Solaris I managed to get a full build. :) patch comming in later :)

File: cmake/GhcHelper.cmake [hunk 1]
```diff
@@ -22,6 +22,9 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND (CMAKE_CXX_COMPILER_VERSION V
     if(${CMAKE_SYSTEM_NAME} MATCHES "(SunOS|Solaris)")
         target_link_libraries(filesystem_test xnet)
     endif()
+    if(${CMAKE_SYSTEM_NAME} MATCHES "(Haiku)")
```
Comment (jmairboeck) [inline cmake/GhcHelper.cmake:25]: The parentheses around Haiku are unnecessary (there aren't multiple alternatives here)

File: cmake/GhcHelper.cmake [hunk 2]
```diff
@@ -22,6 +22,9 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND (CMAKE_CXX_COMPILER_VERSION V
...
+        target_link_libraries(filesystem_test -lnetwork)
```
Comment (jmairboeck) [inline cmake/GhcHelper.cmake:26]: the `-l` prefix can be removed, just the library name (i.e. `network`) is enough.

**Review by jmairboeck — COMMENTED** [review]
Just a few style comments.

They apply multiple times, I didn't repeat them.

File: cmake/GhcHelper.cmake (hunk 2 above)
Comment (Begasus) [reply cmake/GhcHelper.cmake:26]: Thanks on the comments, will check them out 👍

**Comment by Begasus** [comment]
Would be nice if this gets merged before a new release: https://github.com/gulrak/filesystem/issues/164 :)

Created a PR at haikuports to have this available on Haiku: https://github.com/haikuports/haikuports/pull/7865

**Comment by gulrak** [comment]
Looks good to me. If the pipeline goes through, I'll merge it. Thanks for the work!

**Pull request merged**